enterprise/frontend/src/metabase-enterprise/comments/components/CommentEditor/CommentEditor.tsx

```diff
@@ -0,0 +1,168 @@
+  bold: true,
+  italic: true,
+  strikethrough: true,
+} as const;
+
+interface Props {
+  active?: boolean;
+  autoFocus?: boolean;
+  initialContent?: DocumentContent | null;
+  placeholder?: string;
+  readonly?: boolean;
+  onBlur?: (content: DocumentContent, editor: Editor) => void;
+  onChange?: (content: DocumentContent) => void;
+  onSubmit?: (content: DocumentContent) => void;
+}
+
+export const CommentEditor = ({
+  active = true,
+  autoFocus = false,
+  initialContent,
+  placeholder = t`Reply…`,
+  readonly = false,
+  onBlur,
+  onChange,
+  onSubmit,
+}: Props) => {
+  const siteUrl = useSelector((state) => getSetting(state, "site-url"));
+  const [content, setContent] = useState<string | null>(null);
+  const dispatch = useDispatch();
+
+  const extensions = useMemo(
+    () =>
+      [
+        StarterKit.configure({ link: false }),
+        SmartLink.configure({
+          HTMLAttributes: { class: "smart-link" },
+          siteUrl,
+        }),
+        Link.configure({
+          HTMLAttributes: { class: CS.link },
+        }),
+        configureMentionExtension({ dispatch }),
+        !readonly && Placeholder.configure({ placeholder }),
+        !readonly && DisableMetabotSidebar,
+      ].filter(Boolean) as Extension[],
+    [siteUrl, dispatch, readonly, placeholder],
+  );
+
+  const editor = useEditor(
+    {
+      extensions,
+      content: initialContent || "",
+      autofocus: autoFocus,
+      editable: !readonly,
+      immediatelyRender: true,
+      onUpdate: ({ editor }) => {
+        const doc = editor.getText();
+        setContent(doc);
+        if (onChange) {
+          onChange(editor.getJSON() as DocumentContent);
+        }
+      },
+      onBlur: ({ editor }) => {
+        if (onBlur) {
+          onBlur(editor.getJSON() as DocumentContent, editor);
+        }
+      },
+    },
+    [readonly, autoFocus],
+  );
+
+  useEffect(() => {
+    if (editor && initialContent) {
+      editor.commands.setContent(initialContent);
+    }
+  }, [editor, initialContent]);
+
+  if (!editor) {
+    return null;
+  }
+
+  const submitDoc = () => {
+    const content = editor.getJSON() as DocumentContent;
+    const isEmpty = editor.isEmpty;
+    if (!isEmpty && onSubmit) {
+      onSubmit(trimTrailingEmptyParagraphsJSON(content));
+      editor.commands.clearContent(true);
+    }
+  };
+  const handleKeyDown: React.KeyboardEventHandler<HTMLDivElement> = (e) => {
+    if (readonly) {
+      return;
+    }
+    if (e.key === "Enter" && !e.shiftKey) {
+      // see CustomMentionExtension
+      const isMentionOpen = (editor.storage as any)?.mention
+        ?.isMentionPopupOpen;
+      if (isMentionOpen) {
+        return;
+      }
+      e.preventDefault();
+      if (!onSubmit) {
+        return;
+      }
+      submitDoc();
+    }
+  };
+
+  return (
+    <Flex
+      className={cx(S.container, {
+        [S.readonly]: readonly,
+        [S.active]: active,
+      })}
+      onKeyDownCapture={handleKeyDown}
```

Comment:
There must be a way to do this without the capture hack. Mention plugin should control this, and stop propagation of events when appropriate.